lfs/gitfilter_smudge.go

```diff
@@ -16,6 +16,18 @@ import (
 
 func (f *GitFilter) SmudgeToFile(filename string, ptr *Pointer, download bool, manifest *tq.Manifest, cb tools.CopyCallback) error {
 	os.MkdirAll(filepath.Dir(filename), 0755)
+
+	if stat, _ := os.Stat(filename); stat != nil && stat.Mode()&0200 == 0 {
```

Comment:
Since this is being done in the smudge code, is it possible this will affect read-only files in other situations that we're not expecting?

Reply:
I think we should be OK here, since SmudgeToFile is only used to smudge into the working copy, which is the only place that files could be locked or otherwise marked as read-only. The more general-purpose `Smudge()` is what goes out to the process filter and et.

Reply:
Thanks! In that case I am 👍